pkg/stub/handler.go

```diff
@@ -735,6 +690,27 @@ func (h *Handler) CreateDefaultResourceIfNeeded(cfg *v1.Config) (*v1.Config, err
 	return cfg, nil
 }
 
+func (h *Handler) initConditions(cfg *v1.Config) *v1.Config {
+	now := kapis.Now()
+	cfg.Condition(v1.SamplesExist)
+	cfg.Condition(v1.ImportCredentialsExist)
+	valid := cfg.Condition(v1.ConfigurationValid)
+	// our default config is valid; since Condition sets new conditions to false
+	// if we get false here this is the first pass through; invalid configs
+	// are caught above
+	if valid.Status != corev1.ConditionTrue {
+		valid.Status = corev1.ConditionTrue
+		valid.LastUpdateTime = now
+		valid.LastTransitionTime = now
+		cfg.ConditionUpdate(valid)
+	}
```

Comment:
any reason not to just init the conditionvalid to false on the first pass(bootstrap?) and then when we see the CR we'll process the config and (assuming it's valid) update the configvalid to true at that point?

Reply:
as you alluded to, this way saves us a round trip with the API server

this optimization is safe (IMO) since our initially created config is valid

Reply:
k